src/cmd/htmlfmt/html.c

```diff
@@ -22,17 +22,18 @@ loadhtml(int fd)
 	int n;
 	char buf[4096];
 
+	if((n = read(fd, buf, sizeof buf)) <= 0)
```

Comment:
Has this ever been an issue in practice? `htmlfmt` is not meant to be a long-lived program, and when it exits all of its memory will be implicitly freed. The only memory this saves on is a small amount for `u` and `b` and again a small amount in `parsehtml`.

If you want to go down this path, I think I would just stat the file here, and if it's length is 0, `return nil`.  The loop and so forth would not need to change.

Reply:
> The only memory this saves on is a small amount for `u` and `b` and again a small amount in `parsehtml`.

I agree with you.

> If you want to go down this path, I think I would just stat the file here

I think it's better to keep it that way. Thank you for your insight.
Close this PR.